js-core/homeworks/homework-10/src/main.js

```diff
@@ -0,0 +1,138 @@
+/*
+ *
+ * Задача 0
+ *
+ * Что вернет выражение z(x) ?
+ * Напишите ответ своими словами как вы понимаете
+ * В консоле не смотрите, сначала напишите, после проверьте себя
+ *
+ * */
+
+let y = 5;
+
+//создана переменная y и ей присвоенно значение 5
+
+let x = () => y;
+
+//создана переменная x в которую записана функция, возвращающая y
+
+let z = t => {
+  let y = 5;
+  t();
+};
+//console.log(y);
+
+//созданна переменная z которая развна функции t которая создает внутреннею переменную y и вызывает сама себя
+
+console.log(z(x)); // что вернет
+
+//undefined - так как t ничего не возвращает
+
+/*
+ *
+ * TASK 1
+ * Создайте функцию которая будет превращать
+ * передаваемую строку в html тэг
+ *
+ *
+ * */
+
+let $ = tag => `<${tag}></${tag}>`;
+
+let createBODY = $("body");
+let createDIV = $("div");
+console.log(createBODY); // <body></body>
+console.log(createDIV); // <div></div>
+
+/*
+ *
+ * TASK 2
+ *
+ * Создайте объект к которому можно будет применить любое число вызовов
+  // obj.method().method().method()
+  ---------------
+ *  Передаваемое значение должно возвращаться в виде html тэгов (TASK 1)
+ *  Передаваемые аргументы должны быть только в виде строки
+ * */
+
+var ezjQueryBase = {
+  tags: "",
+  add: function(str) {
+    ezjQueryBase.tags = ezjQueryBase.tags + `<${str}></${str}>`;
+    console.log(ezjQueryBase.tags);
+    return this;
+  }
+};
+
+ezjQueryBase
+  .add("body") // <body></body>
+  .add("div") // <body></body><div></div>
+  .add("h1"); // <body></body><div></div><h1></h1>
+
+/*
+ *
+ * TASK 3
+ * Доработйте метод add чтобы на каждом вызове следующий
+ * тэг помещался внутри предыдущего !
+ ---
+ * И добавьте объекту ezjQuery метод render, который будет возвращать
+ * сгенерированную строку
+ -----
+ * Методу add - второй параметр, который будет размещать
+ * информацию внутри тэга
+ *
+ */
+var ezjQuery = {
+  tags: "",
+  add: function(...arg) {
+    let tag = [...arg][0];
```

Comment:
this is no good practice... let take a look what happens here
 
at line 87 you get new variable *arg* with type Array.
at line 88 you again apply destruction to variable arg and make in Array again...
and then you receive the first argument.

Why this bad, because it's not obvious what you want to receive, just direct access to props will be fine